main_objs.py

```diff
@@ -86,26 +86,46 @@ def __init__(self, score):
             ('c', True, True): lambda cell: str(cell.semitones) if hasattr(cell, 'semitones') else cell,
             ('c', True, False): lambda cell: str(cell.semitones % 12) if hasattr(cell, 'semitones') else cell,
             ('c', False, True): lambda cell: str(abs(cell.semitones)) if hasattr(cell, 'semitones') else cell,
-            ('c', False, False): lambda cell: str(abs(cell.semitones)) % 12 if hasattr(cell, 'semitones') else cell
+            ('c', False, False): lambda cell: str(abs(cell.semitones) % 12) if hasattr(cell, 'semitones') else cell
         }
 
     def _getPartSeries(self):
         if 'PartSeries' not in self.analyses:
-            parts = self.score.getElementsByClass(stream.Part)
             part_series = []
-            for i, part in enumerate(parts):
-                notesAndRests = part.flat.getElementsByClass(['Note', 'Rest'])
-                part_name = part.partName or 'Part_' + str(i + 1)
-                ser = pd.Series(notesAndRests, name=part_name)
+
+            for i, flat_part in enumerate(self._getFlatParts()):
+                notesAndRests = flat_part.getElementsByClass(['Note', 'Rest'])
+                ser = pd.Series(notesAndRests)
                 ser.index = ser.apply(lambda noteOrRest: noteOrRest.offset)
                 ser = ser[~ser.index.duplicated()] # remove multiple events at the same offset in a given part
                 part_series.append(ser)
             self.analyses['PartSeries'] = part_series
         return self.analyses['PartSeries']
 
+    def _getFlatParts(self):
+        """
+        Return and store flat parts inside a piece using the score attribute.
+        """
+        if 'FlatParts' not in self.analyses:
+            parts = self.score.getElementsByClass(stream.Part)
+            self.analyses['FlatParts'] = pd.Series([part.flat for part in parts])
```

Comment:
We don't need this to be a series here, so you can just make it:
self.analyses['FlatParts'] = [part.flat for part in parts]